evalml/automl/automl_search.py

```diff
@@ -522,9 +527,46 @@ def search(self, data_checks="auto", show_iteration_plot=True):
 
         best_pipeline = self.rankings.iloc[0]
         best_pipeline_name = best_pipeline["pipeline_name"]
+        self._best_pipeline = self.get_pipeline(best_pipeline['id'])
+        if self._train_best_pipeline:
+            X_train = _convert_to_woodwork_structure(X)
+            y_train = _convert_to_woodwork_structure(y)
```

Comment:
I think you should get rid of this now that #1597 is merged, and call `self.X_train` / `self.y_train`

Reply:
I wrote this before you rebased/updated your PR, so LGTM!